controllers/clustergroupupgrade_controller.go

```diff
@@ -2112,6 +2115,19 @@ func (r *ClusterGroupUpgradeReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Version: "v1",
 	})
 
+	maxConcurrency, set := os.LookupEnv(utils.CGUControllerWorkerCountEnv)
+	var maxConcurrentReconciles int
+	var err error
+	if set {
+		maxConcurrentReconciles, err = strconv.Atoi(maxConcurrency)
+		if err != nil {
```

Comment:
Shouldn't we also check for maxConcurrentReconciles to be > 0?

Reply:
Added